extensions/consul-config/runtime/src/main/java/io/quarkus/consul/config/runtime/ConsulConfigGateway.java

```diff
@@ -1,12 +1,14 @@
 package io.quarkus.consul.config.runtime;
 
-import java.io.IOException;
 import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
 
 interface ConsulConfigGateway {
 
     /**
      * Retrieves a value from Consul's Key / Value store using the value of {@code key}
      */
-    Optional<Response> getValue(String key) throws IOException;
+    CompletableFuture<Optional<Response>> getValue(String key);
```

Comment:
Why a CompletionStage instead of a Uni? 

Also, we rarely use CompletionStage<Optional<X>>, `null` is part of the language.

Reply:
Makes sense, I will update soon